sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/implementation/models/LifetimeActionTrigger.java

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.security.keyvault.keys.implementation.models;
+
+import com.azure.core.annotation.Fluent;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * A condition to be satisfied for an action to be executed.
+ */
+@Fluent
+public final class LifetimeActionTrigger {
+    @JsonProperty(value = "timeAfterCreate")
+    private String timeAfterCreate;
+
+    @JsonProperty(value = "timeBeforeExpiry")
+    private String timeBeforeExpiry;
```

Comment:
As we discussed there might be a better type for this - `Duration` maybe or whatever servicebus queue is using. Whatever is idiomatic for Java to represent ISO8601 Durations.

I also know that this isn't the publicly exposed class but just calling it out here 👍 

Same for timeAfterCreate and timeBeforeExpiry

Reply:
I am refraining of using `Duration` for the time being since it doesn't play nice with durations longer than days. The alternative `Period` class is not meant for anything other than days/months/years. I'd rather keep using a String for now and hear what our architects have to say about this when we do the API review.